frontend/src/app/cities-upload/cities-upload.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component } from '@angular/core';
+import { SearchCityService } from '../search-city.service';
+import { HttpEventType } from '@angular/common/http';
+import { Observable } from "rxjs/Observable";
+import 'rxjs/add/observable/interval';
+import 'rxjs/add/operator/finally';
+import 'rxjs/add/operator/take';
+import 'rxjs/add/operator/takeWhile';
+
+const ESTIMATED_PROCESSING_TIME_IN_MILLIS = 20000;
```

Comment:
Out of curiosity: couldn't we do something with Spring reactive support to have a real stream as a response of the processing process instead of "guessing it"? Not that it is important in that case, just wondering if that could be done more accurately.